src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/TelekineticDefensiveSpell.java

```diff
@@ -0,0 +1,199 @@
+                double velocity = entity.getDeltaMovement().lengthSqr();
+                reduceMana((int)(Math.ceil(velocity * 100 * config.MANA_COST.get())), player);
+
+                entity.move(MoverType.PLAYER, getShift(player, usedHand, entity));
+                entity.resetFallDistance();
+
+                // Handle collision damage
+                // Kinda glitchy, keep off by default
+                if(config.DO_DAMAGE.get()){
+                    float dmg = (float)(entity.getDeltaMovement().lengthSqr() * 100 * config.DAMAGE.get());
+                    if(dmg >= 1 && (entity.horizontalCollision || entity.verticalCollision)){
+                        entity.hurt(DamageSource.playerAttack(player).setIsFall(), dmg);
+                    }
+                }
+
+                entity.setDeltaMovement(Vec3.ZERO);
+            } else {
+                selectedMap.remove(player.getId());
+            }
+        }
+    }
+
+    @Override
+    public void releaseUsing(ItemStack stack, InteractionHand usedHand, Level level, Player player) {
+        InteractionHand cacheHand = attackMap.remove(player.getId());
+        if(cacheHand == null || cacheHand != usedHand){
+            Integer id = selectedMap.remove(player.getId());
+            if(id != null){
+                Entity e = level.getEntity(id);
+                if(e != null && e.isAlive() && e instanceof LivingEntity entity)
+                    entity.setDeltaMovement(getShift(player, usedHand, entity));
+            }
+        }
+    }
+
+    protected Vec3 getShift(Player player, InteractionHand usedHand, LivingEntity entity){
+        Vec3 view = player.getViewVector(0);
+                
+        float yRot = player.getYRot();
+        double x = 0 - Math.sin(yRot * Math.PI / 180f);
+        double z = Math.cos(yRot * Math.PI / 180f);
+        int hand = usedHand == InteractionHand.MAIN_HAND ? 1 : -1;
+        Vec3 playerPos = new Vec3(
+                player.getX() - (z / 2.5d * hand),
+                player.getEyeY() - 1d,
+                player.getZ() + (x / 2.5d * hand)
+        );
+        return view.scale(config.HOLD_DIST.get()).add(playerPos).subtract(entity.position());
+    }
+
+    @Override
+    public void appendHoverText(ItemStack pStack, Level pLevel, List<Component> pTooltipComponents, TooltipFlag pIsAdvanced) {
+        super.appendHoverText(pStack, pLevel, pTooltipComponents, pIsAdvanced);
+        pTooltipComponents.add(new TextComponent("When activated:").withStyle(ChatFormatting.GRAY));
+        pTooltipComponents.add(new TextComponent("  - Picks up the currently looked at mob from up to ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent(config.GRAB_DISTANCE.get() + " blocks away").withStyle(ChatFormatting.WHITE))
+                .append(" and moves them where the player is looking from a distance of ")
+                .append(new TextComponent(config.HOLD_DIST.get() + " blocks away").withStyle(ChatFormatting.WHITE)));
```

Comment:
Use `ChatFormatting.DARK_GREEN` for the block color highlighting since that's what we've been using everywhere else.

Reply:
fixed